activerecord/test/cases/connection_adapters/connection_handler_test.rb

```diff
@@ -16,6 +16,19 @@ def setup
         @pool = @handler.establish_connection(ActiveRecord::Base.configurations["arunit"])
       end
 
+      def test_establish_connection_does_not_leak_pools
+        # The pool we want to be GC'd
+        pool = @handler.establish_connection(:arunit)
+        original_ref = WeakRef.new(pool)
+
+        # Establish a new connection and drop the local variable reference
+        pool = @handler.establish_connection(:arunit)
+
+        GC.start
+
+        assert_not original_ref.weakref_alive?
```

Comment:
```suggestion
        assert_not_predicate original_ref, :weakref_alive?
```

Reply:
I removed the test since it [seemed flaky on CI](https://buildkite.com/rails/rails/builds/61330)